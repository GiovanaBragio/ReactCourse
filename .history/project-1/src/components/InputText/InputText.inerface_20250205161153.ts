export interface InputTextProps {
    value: string;
    onChange: () => void;
}