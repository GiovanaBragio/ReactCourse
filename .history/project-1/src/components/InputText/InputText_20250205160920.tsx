import { InputTextProps } from "./InputText.inerface";

const InputText = ({ value, onChange }: InputTextProps) => {
    return (
        <input
            type="text"
            placeholder="Search"
            value={value}
            onChange={onChange} />
    )
}

export default InputText;