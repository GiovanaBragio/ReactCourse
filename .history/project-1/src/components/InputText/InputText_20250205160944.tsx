import { InputTextProps } from "./InputText.inerface";

const InputText = ({ value, onChange }: InputTextProps) => {
    return (
        <input
            className="inputText"
            type="text"
            placeholder="Search"
            value={value}
            onChange={onChange} />
    )
}

export default InputText;