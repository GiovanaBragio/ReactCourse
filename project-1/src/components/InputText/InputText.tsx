import { InputTextProps } from "./InputText.inerface";
import './InputText.styles.css';

const InputText = ({ value, onChange }: InputTextProps) => {
    return (
        <input
            className="inputText"
            type="text"
            placeholder="Search"
            value={value}
            onChange={onChange} />
    )
}

export default InputText;