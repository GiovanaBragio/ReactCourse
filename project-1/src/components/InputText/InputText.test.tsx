import { render, screen } from "@testing-library/react";
import InputText from "./InputText";
import userEvent from "@testing-library/user-event";

describe('InputText', () => {
  it('should have a value of searchValue', () => {
    render(<InputText onChange={() => { }} value={'testing'} />);
    const input = screen.getByPlaceholderText("Search") as HTMLInputElement;
    
    expect(input.value).toBe('testing');
    
  });
  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<InputText onChange={fn} value={'testing'} />);
    const input = screen.getByPlaceholderText("Search") as HTMLInputElement;
    const value = 'testing';
    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });
  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<InputText onChange={fn} value={'testing'} />);
    expect(container).toMatchSnapshot();
  });
});