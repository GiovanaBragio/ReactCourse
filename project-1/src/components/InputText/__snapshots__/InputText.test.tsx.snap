// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should match snapshot 1`] = `
<div>
  <input
    class="inputText"
    placeholder="Search"
    type="text"
    value="testing"
  />
</div>
`;
